### DISCUSSION TITLE
command 'github.copilot.generate' not found

### DISCUSSION THREAD

**Discussion by addhyay** [community/community#6965] [Copilot Conversations]
I am trying to install GitHub copilot on VScode but after installation when I use it then it says, command 'github.copilot.generate' not found. what should I do?

![Screenshot 2021-10-29 080728](https://user-images.githubusercontent.com/46434438/139364490-1726a57e-391d-497c-b34e-9a3625b86417.png)

**Comment by davecheney** [comment]
that may be intercepting your HTTPS connections?

**Reply by addhyay** [reply]
can you elaborate, please?

**Reply by davecheney** [reply]
I’m sorry, my answer was truncated. What I meant to ask is there a firewall, vpn, or other device that may be interceding on your network traffic.

**Reply by addhyay** [reply]
no, not at all

**Reply by davecheney** [reply]
Thanks for confirming. We've found that copilot will have difficulties starting up if it cannot connect to our servers. I believe there is a new version of the extension coming soon which will provide more detail when it is unable to connect to our servers.

**Reply by addhyay** [reply]
Thank you so much! :)

**Reply by sahilbakoru** [reply]
![Screenshot 2021-11-11 at 8 25 48 PM](https://user-images.githubusercontent.com/68190549/141319607-00470f89-189d-449d-b201-3acc8850190d.png)

it still has the same problem , i restarted , reinstall, i re login , but its still not working .

can anybody help ?

**Comment by jamaldevsnet** [comment]
Hi @davecheney I am having the same issue. While looking at the copilot output is says

`[INFO] [auth] [2021-11-02T01:39:48.245Z] Invalid copilot token: missing token`

Could you please help me?

**Reply by davecheney** [reply]
@jamaldevsnet looks like you're using the intellij plugin. Is that correct? /cc @jansorg

**Reply by jamaldevsnet** [reply]
Hey @davecheney I am using vscode. I have tried with a fresh installation of vscode with no extensions except copilot, still no luck!

**Reply by davecheney** [reply]
Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by jamaldevsnet** [reply]
@davecheney I am not under any firewall. I'm running ubuntu, even the ufw is disabled. I am not using VPN either.

**Reply by jamaldevsnet** [reply]
@davecheney Just configured copilot with neovim, it's working fine there. But still no luck with vscode :(

**Reply by davecheney** [reply]
this sounds like an authentication problem between vscode at github to get a token. the process is somewhat more automated than neovim, but I don’t know the details of how to deauthenticate and authenticate again, sorry. Try searching on this forum, or I think a lot of discussion is happening on stack overflow.

**Reply by jamaldevsnet** [reply]
@devoncurry I will see if I can find something. Thanks for your insights into the issue, much appreciated :smiley:

**Reply by davecheney** [reply]
perhaps https://github.com/github/feedback/discussions/6711#discussioncomment-1534756

**Comment by ghost** [comment]
same here.. command `github.copilot.generate` not found. Did all the voodoo suggested across internet and nothing worked.

**Reply by davecheney** [reply]
Hello, I'm sorry you are having difficulties. We've found that the copilot icon may not show up if copilot is not able to contact our servers. Are you behind a firewall, vpn, or other technology that may be intercepting your network connections?

**Reply by ghost** [reply]
> same here.. command `github.copilot.generate` not found. Did all the voodoo suggested across internet and nothing worked.

Resolved after sign-out and signing back in.

**Reply by Calvett** [reply]
Same problem for me, initially everything was fine, Copilot was working fine, but this morning when I turned on my laptop and launched VSCode, the Copilot symbol was missing, I tried everything I read:
- uninstall and reload, reinstall -> same error
- uninstall and reload, log out of GitHub, reinstall, log in -> same error
- uninstall and reload, log out of GitHub, delete folder ".vscode / extensions / github.copilot-1.7.3689", reinstall, log in -> same error

There are some people commenting on a wrong or not accepted authentication flow correctly, but it was fine on Friday and I haven't even opened the laptop until today ... :(

**Reply by jamal-0x1** [reply]
I have tried removing and reinstalling but didn't work. I had to set up the OS to fix it lol.

**Reply by Calvett** [reply]
The new update of copilot extension has change the error messege, now is:
GitHub Copilot could not connect to server. Extension activation failed: "self signed certificate in certificate chain"
baby steps? :)

**Reply by davecheney** [reply]
That suggests someone is MITMing your SSL traffic.

**Reply by Calvett** [reply]
It's a company laptop but i'm not connect through VPN or similar it's a personal network, but I don't know if there are some tools install that prevent Github login into the server. The bizarre thing is that the extension works on instalation but only the first day. So I could taste it and never used again like in a greek tragedy :)

**Reply by jamal-0x1** [reply]
Try switching to a different network.

**Reply by gj1118** [reply]
@Calvett were you manage to resolve this issue please ? I am stuck in the same shoes as yourself , so if you have been able to resolve it can you please let me know about it as well ? Thanks ..

**Reply by andigwandi** [reply]
same here, using the company's laptop and stuck at the same error. I tried it on a Citrix machine and it worked fine with my account.

**Reply by davecheney** [reply]
@andigwandi please upgrade to a more recent version of copilot, the error you are mentioning has long been solved.
